src/Radicle/Ipfs.hs

```diff
@@ -38,33 +38,54 @@ import           Data.Conduit ((.|))
 import qualified Data.Conduit as C
 import qualified Data.Conduit.Attoparsec as C
 import qualified Data.Conduit.Combinators as C
+import qualified Data.HashMap.Strict as HashMap
 import           Data.IPLD.CID
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
 import           Lens.Micro ((.~), (^.))
 import           Network.HTTP.Client
                  (HttpException(..), HttpExceptionContent(..))
 import qualified Network.HTTP.Client as HTTP
+import qualified Network.HTTP.Client as Http
 import qualified Network.HTTP.Conduit as HTTP
 import qualified Network.Wreq as Wreq
 import           System.Environment (lookupEnv)
 
 data IpfsException
-    = IpfsException Text
-    -- | JSON response from the IPFS Api cannot be parsed. First
-    -- argument is the request path, second argument the JSON parsing
-    -- error
-    | IpfsExceptionInvalidResponse Text Text
-    -- | The IPFS daemon is not running.
-    | IpfsExceptionNoDaemon
-    deriving (Show, Eq)
+  = IpfsException Text
+  | IpfsExceptionErrResp Text
+  | IpfsExceptionErrRespNoMsg
+  -- | The request to the IPFS daemon timed out.
+  | IpfsExceptionTimeout
+  -- | JSON response from the IPFS Api cannot be parsed. First
+  -- argument is the request path, second argument the JSON parsing
+  -- error
+  | IpfsExceptionInvalidResponse Text Text
+  -- | The IPFS daemon is not running.
+  | IpfsExceptionNoDaemon
+  deriving (Show)
 
 instance Exception IpfsException where
-    displayException (IpfsException msg) = "ipfs: " <> toS msg
-    displayException IpfsExceptionNoDaemon = "ipfs: Daemon not reachable, run 'rad ipfs daemon"
-    displayException (IpfsExceptionInvalidResponse url _) = "ipfs: Cannot parse IPFS daemon response for " <> toS url
-
-
+    displayException = \case
+      IpfsException msg -> "ipfs: " <> toS msg
+      IpfsExceptionNoDaemon -> "ipfs: Daemon not reachable, run 'rad ipfs daemon'"
```

Comment:
The command is not correct anymore. Since most people are going to use systemd to run the deamon this might also be confusing. Maybe we could replace the message with "Cannot connect to Radicle IPFS daemon".